lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -451,18 +451,21 @@ public function add($sqlPartName, $sqlPart, $append = false)
      * Specifies an item that is to be returned in the query result.
      * Replaces any previously specified selections, if any.
      *
+     * USING AN ARRAY ARGUMENT IS DEPRECATED. Pass each value as an individual argument.
+     *
      * <code>
      *     $qb = $conn->createQueryBuilder()
      *         ->select('u.id', 'p.id')
      *         ->from('users', 'u')
      *         ->leftJoin('u', 'phonenumbers', 'p', 'u.id = p.user_id');
      * </code>
      *
-     * @param mixed $select The selection expressions.
+     * @param string|string[]|null $select The selection expression. USING AN ARRAY OR NULL IS DEPRECATED.
+     *                                     Pass each value as an individual argument. Pass at least one value.
```

Comment:
> Pass at least one value.

This is not necessary. You still will be able to call it without arguments to declare a `SELECT` query. At least it’s the current state.

Reply:
Hmm, true. Actually I think we should force at least one value in `select()` as well; but we can keep this for another time.

Reply:
I deliberately kept it like for similarity with `insert()`, `update()` and `delete()`. It just defines the query type. Definitely worth revisiting later.